src/dls_bluesky_core/core/maths.py

```diff
@@ -0,0 +1,47 @@
+from typing import Tuple
+
+import numpy as np
+
+
+def step_to_num(start: float, stop: float, step: float) -> Tuple[float, float, int]:
+    """
+    Standard handling for converting from start, stop, step to start, stop, num
+    Forces step to be same direction as length
+    Includes a final point if it is within 1% of the end point (consistent with GDA)
+
+    Args:
+        start (float):
+            Start of length, will be returned unchanged
+        stop (float):
+            End of length, if length/step does not divide cleanly will be returned
+            extended up to 1% of step, or else truncated.
+        step (float):
+            Length of a step along the line formed from start to stop.
+            If stop < start, will be coerced to be backwards.
+
+    Returns:
+        start, truncated_stop, num = Tuple[float, float, int]
+        start will be returned unchanged
+        truncated_stop = start + num * step
+        num is the maximal number of steps that could fit into the length.
+
+    """
+    # Make step be the right direction
+    step = abs(step) if stop >= start else -abs(step)
+    # If stop is within 1% of a step then include it
+    steps = int((stop - start) / step + 0.01)
+    return start, start + steps * step, steps + 1  # include 1st point
+
+
+def in_micros(t: float) -> int:
+    """
+    Converts between units of microT and units of T.
+    For example, from microseconds to seconds.
+
+    Args:
+        t (float): A time in microseconds, or other measurement in units of microU
+    Returns:
+        t (int): A time in seconds rounded up to the nearest whole second,
+        or other measurement in units of U, rounded up to the nearest whole U.
+    """
+    return int(np.ceil(t / 1e6))
```

Comment:
Oh dear, I wrote this wrong originally, can you tell I never ran the code?

```suggestion
def in_micros(seconds: float) -> int:
    """Convert a time in seconds to an integer number of microseconds, rounding up
    
    Args:
        seconds: A time in seconds
    Returns:
        A time in microseconds rounded up to the nearest whole microseconds,
    """
    return int(np.ceil(t * 1e6))
```

Reply:
I wasn't sure, given the name of it, but I thought I'd write docs for what it did and if that's not what it's supposed to do that's good :D

Reply:
Will resolve locally as need to change tests